File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        
        //We have used left -> l and right -> r to denote the first pointer and second pointer respectively
        int l = 0;
        int r = numbers.length - 1;
        //
        int[] arrayNum = {l , r}; // or int[] arrayNum = new int[2];
        
        while (l < r) {
            int cursum = numbers[l] + numbers[r];
            
        //When the sum is equal to the target: -> We have found the answer and simply return the required indexes of the above used two pointers.    
            if (cursum == target) {
                arrayNum[0] = l + 1;
                arrayNum[1] = r + 1;
                return arrayNum;
            } else if (cursum > target) { 
        //When the sum is bigger than target: -> We'll move right pointer to the left by 1
                r--;
            } else if (cursum < target) {
        //When the sum is smaller than the target: -> We'll move left pointer to the right by 1        
                l++;
            }
        } return arrayNum;
    }
}

File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        #We have used left -> l and right -> r to denote the first pointer and second pointer respectively
        l = 0
        r = len(numbers) - 1
        while(l < r):
            #When the sum is equal to the target: -> We have found the answer and simply return the required indexes of the above used two pointers.
            if numbers[l] + numbers[r] == target:
                return [l+1, r+1]
            #When the sum is bigger than target: -> We'll move right pointer to the left by 1
            elif numbers[l] + numbers[r] > target:
                r = r - 1
            else:
                l = l + 1
        
        
